### DISCUSSION TITLE
RevelJS: How to create a horizontal line between slide content and footnotes

### DISCUSSION THREAD

**Discussion by jkub6** [quarto-dev/quarto-cli#8534] [Q&A] [answered]
### Description

I really like the look of the horizontal line between the footnotes and the rest of the slide I get when setting "reference-location: section", but this messes up the footnote numbering (see slide two). I imagine I am probably using this setting in an unintended way on a RevealJS slide. Is there another way to get a horizontal line in this position? (only on slides with footnotes)

## Example:

![image](https://github.com/quarto-dev/quarto-cli/assets/30987328/57f507dc-80b6-42c5-8481-bf98ef954239)

![image](https://github.com/quarto-dev/quarto-cli/assets/30987328/737fc909-dde1-4888-bbd3-11635a085190)

````qmd
---
title: test
format:
  revealjs:
    center: true
reference-location: section
---

## Slide 1

- bullet one
- bullet two^[first footnote]
- This is the third bullet

## Slide 2

more test text here^[second footnote]
````

**Comment by mcanouil** [comment, marked as answer]
You can open your slide deck in your favourite browser, inspect the slide around the footnotes, then target the (S)CSS class and add the appropriate elements to add a line above the footnotes "section".

Note that "section" in Reveal.js does not have the same meaning as in standard HTML document.
Section means first level headings, so it is normal to have footnotes count increasing between two level 2 header slides.
What were you expected this to be doing in the context of Reveal.js exactly? There might be something to be done.

Edit:

````qmd
---
title: test
format:
  revealjs:
    center: true
include-in-header:
  text: |
    <style type="text/css">
      .reveal .slide aside {
        border-top: 1px solid;
        border-style: inset;
        padding-top: 1em;
      }
    </style>
---

## Slide 1

- bullet one
- bullet two^[first footnote]
- This is the third bullet^[second footnote]

## Slide 2

more test text here^[third footnote]
````

**Reply by jkub6** [reply]
That works great, thanks! I wasn't expecting anything in particular, just playing with the options.

**Reply by mcanouil** [reply]
CC @cderv in case you see something to be tweaked in how the `reference-location` works for `revealjs` format inside Quarto.

**Reply by cderv** [reply]
> something to be tweaked in how the reference-location works for revealjs format inside Quarto.

Maybe it should be disallowed for revealjs ? Not sure to see what really this should mean in the context of revealjs... 

Footnotes are already on their slide right ?

**Reply by mcanouil** [reply]
Yes
`reference-location: section` is kind of the `revealjs` default.
